packages/react-native/ios/BugsnagReactNative/BugsnagEventDeserializer.m

```diff
@@ -98,6 +100,21 @@ - (BugsnagEvent *)deserializeEvent:(NSDictionary *)payload {
         event.errors[0].errorMessage = error[@"errorMessage"];
         [event attachCustomStacktrace:error[@"stacktrace"] withType:@"reactnativejs"];
     }
+    
+    id nativeStack = payload[@"nativeStack"];
+    if ([nativeStack isKindOfClass:[NSArray class]] &&
+        [nativeStack filteredArrayUsingPredicate:
+         [NSPredicate predicateWithFormat:@"NOT SELF isKindOfClass: %@", [NSString class]]].count == 0) {
+        NSArray<BugsnagStackframe *> *stackframes = [BugsnagStackframe stackframesWithCallStackSymbols:nativeStack];
+        if (stackframes != nil) {
+            BugsnagError *nativeError = [BugsnagError new];
```

Comment:
The `type` needs to be set to `BSGErrorTypeCocoa` as it's ultimately used by the [Error Reporting API](https://bugsnagerrorreportingapi.docs.apiary.io/#reference/0/notify/send-error-reports) to ensure that stacktraces render correctly for each platform. Using the `initWithErrorClass:errorMessage:errorType:stacktrace:` construct would guarantee that all properties on the object are initialized correctly.